sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-factory.ts

```diff
@@ -0,0 +1,342 @@
+}
+
+/**
+ * Creates and saves the students populated from the test input data.
+ * @param db data source helper.
+ * @param testInputStudent data to create the student, its user, and
+ * associated SIN validation.
+ * @returns saved student.
+ */
+async function saveIER12StudentFromTestInput(
+  db: E2EDataSources,
+  testInputStudent: IER12Student,
+): Promise<Student> {
+  // User
+  const fakeUser = createFakeUser();
+  fakeUser.firstName = testInputStudent.firstName;
+  fakeUser.lastName = testInputStudent.lastName;
+  // Student
+  const fakeStudent = createFakeStudent(fakeUser);
+  fakeStudent.birthDate = getISODateOnlyString(testInputStudent.birthDate);
+  fakeStudent.contactInfo = {
+    address: {
+      ...testInputStudent.addressInfo,
+      // The country and the selectedCountry variables are not part of IER12, hence keeping as constants.
+      country: "canada",
+      selectedCountry: "Canada",
+    },
+    phone: faker.phone.phoneNumber(),
+  };
+  // SIN validation
+  const sinValidation = createFakeSINValidation({ student: fakeStudent });
+  sinValidation.sin = testInputStudent.sin;
+  // Student
+  return saveFakeStudent(db.dataSource, {
+    student: fakeStudent,
+    sinValidation,
+  });
+}
+
+/**
+ * Creates and saves the application populated from the test input data
+ * alongside with its disbursements.
+ * @param db data source helper.
+ * @param testInputApplication data to create the application and its dependencies.
+ * @param student previously saved student.
+ * @param institutionLocation previously created institution location.
+ * @param programYear previously saved program year.
+ * @param createSecondDisbursement indicates if a second disbursement should be created.
+ * @param referenceSubmission date when the application was submitted.
+ * @returns the saved applications and its dependencies.
+ */
+async function saveIER12ApplicationFromTestInput(
+  db: E2EDataSources,
+  testInputApplication: IER12Application,
+  student: Student,
+  institutionLocation: InstitutionLocation,
+  programYear: ProgramYear,
+  createSecondDisbursement: boolean,
+  referenceSubmission: Date,
+): Promise<Application> {
+  const application = await saveFakeApplicationDisbursements(
+    db.dataSource,
+    { student, institutionLocation, disbursementValues: [] },
+    { createSecondDisbursement },
+  );
+  application.applicationNumber = testInputApplication.applicationNumber;
+  application.studentNumber = testInputApplication.studentNumber;
+  application.relationshipStatus = testInputApplication.relationshipStatus;
+  application.submittedDate =
+    testInputApplication.submittedDate ?? referenceSubmission;
+  application.applicationStatus = testInputApplication.applicationStatus;
+  application.applicationStatusUpdatedOn =
+    testInputApplication.applicationStatusUpdatedOn ?? referenceSubmission;
+  application.programYear = programYear;
+  await db.application.save(application);
+  return application;
```

Comment:
maybe instead of lines 194 and 195, `return db.application.save(application);`?

Reply:
Changed.